### DISCUSSION TITLE
Command Block Staff functionality concept

### DISCUSSION THREAD

**Discussion by Indying** [opekope2/StaffMod#51] [Ideas]
AVM's command block obviously defies anything possible within reason, but that doesn't mean that it has to be a portable command block and nothing more, with a few keybinds it could be a pretty cool item.

**_Would it even be obtainable?_**

Possibly, though I doubt it. If it were craftable, it'd be through a recipe using probably a dragon egg, nether star, and some other rare stuff, allowing you to make a "Keyboard-less Command Block", AKA: An excuse for the command block to not function normally :p

**_What would the Command Block Staff do?_**

It would have various keybinds that do some of the notable things from AVM, like...

_**Basic Uses**_

- Custom Keybind - Pick up and move around entity
- Shift + Scroll - Move entity away/towards you
- Shift + Right Click - Form a Shield around you
- Shift + Scroll with shield active - Increase/Decrease shield radius
- Keybind - Activate/Deactivate Hover/Flight mode

_**Fill Command**_

- Keybind - Brings up a menu with 5 fields to input blocks, these would be quickly selectable blocks that you could cycle through
- Keybind - Cycle through quick-select blocks
- Right Click air - Place block, at default 3 blocks away from you
- Shift + Scroll - Cycle distance. Min = 1 block away, Max = 10 blocks away

Might be a bit difficult to code, but the entire idea of adding the command block staff is kinda insane, anyway tell me what you think 👍

**Comment by opekope2** [comment]
The "Keyboard-less Command Block" is a good idea, this would explain why players can't run just any command.

I started to work on the Command Block Staff, and it'll be able to do much more than just a few commands. I saw a couple of videos by @Pandzio21, which showcased some of the ideas you mentioned.

I opened a discussion to discuss the Command Block Staff in one place: https://github.com/opekope2/StaffMod/discussions/86

**Comment by Pandzio21** [comment]
Yeah @Indying cool you mention some of my ideas like the flight, shield but i don't think flight will be a good option because in the series you couldn't fly with the command block Staff and i wanted to make some changes to the shield but as of now im not implementing features to the actual official  mod but modifying my own modified version of an older build of opekope2's mod but i think i would make the shield not made of particles but maybe a model idk yet still didnt fully thought of that part and the update that will be made is not gonna be really the same as in my videos

**Reply by opekope2** [reply]
Actually, Yellow did fly with the command block staff: https://youtube.com/shorts/LDJCMHqISfw
